### ISSUE TITLE
Tests: Refactor lib/plugins/aws/deploy/lib/checkForChanges.test.js 

### ISSUE THREAD

**Issue by medikoo** [serverless/serverless#8886]
_Most of the tests that currently cover core functionalities depend heavily on (and sometimes test) internal implementation characteristics. While they should be testing whether given implementation produces desired outcome (treating its implementation more as  a black box)._

_Current state of things is problematic for eventual internal improvements and refactors which occasionally we want to introduce, as in most of such cases proposed improvements need to be accompanied with counterproductive numerous updates to tests which are covering otherwise not altered functionalities ._

_At some point we've introduced a new (black box based) way of testing the internals. It's through `runServerless` utlity, which allows us create natural (as in real world) `serverless` instance, and inspect the produced outcome for chosen command. More details here: https://github.com/serverless/serverless/tree/master/test#unit-tests_

--- 

This is issue is about refactoring [test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js ](https://github.com/serverless/serverless/blob/7c998e03ca104948c6c2673c9fee6992689219df/test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js) to `runServerless` based variant.

To make refactor relatively easy:
- Spec for new tests is fully defined and is already committed in. It's in scope of this _describe_ block: https://github.com/serverless/serverless/blob/7c998e03ca104948c6c2673c9fee6992689219df/test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js#L943-L1564
- Each test slot indicates which test it is about to replace
- `runServerless` runs are already preconfigured, with comments indicating where blank dots need to be filled

For every refactored test, the old test should be removed

Refactor can be submitted with one PR, but partial refactors that cover some of tests are also very welcome!

**Comment by AlinoeDoctari** [comment]
I've started to have a look, more specifically on the checkLogGroupSubscription refactor and I have a bit of trouble making the proposed implementation work. It seems the test framework does not expose a beforeInstanceInit hook. Looking into the test code, nor `before` or `after` would do the trick here. What would be the way forward to make that implementation work?

**Comment by medikoo** [comment]
@AlinoeDoctari yes, I totally forgot to add implementation of that hook, give me a moment

**Comment by medikoo** [comment]
@AlinoeDoctari it was just published with `@serverless/test` v7.6.0, ensure to have that dependency up to date, and it should work.

Thanks for pointing!

**Comment by PreetSangha** [comment]
Hi I'm just a novice here. Can you help me understand why this issue is still open when it's marked as merged, please?
